### PR TITLE
Add Chinese language.

### PR DESCRIPTION
**Pull request by huacnlee**
1. Add language `zh-CN`;
2. Support HTTP Header: `HTTP_ACCEPT_LANGUAGE` to set default language;
3. Store locale set in session when use `params[:locale]`, and use it for next request;
4. Add rails-i18n for provide more translate for Rails.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -60,4 +63,16 @@ def find_rubygem
   def set_page
     @page = [1, params[:page].to_i].max
   end
+
+  def user_locale
+    params[:locale] || session[:locale] || http_head_locale  || I18n.default_locale
+  end
+
+  def http_head_locale
+    lang = request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z\-]{5}/i).first
```
Comment (simi) [inline app/controllers/application_controller.rb]: What about to use https://github.com/iain/http_accept_language here?

Comment (huacnlee) [inline app/controllers/application_controller.rb]: :+1:

**Comment by qrush** [comment]
Wow that's great! Is there anything left here or is this good to merge?

File: config/locales/nl.yml [hunk 2]
```diff
@@ -1,7 +1,10 @@
 nl:
   update: Wijzig
+  sign_in: "Sign in"
+  sign_up: "Sign up"
```
Comment (simi) [inline config/locales/nl.yml]: Do we need english locales in `nl.yml`? Is there any chance to reuse I18n fallbacks to fallback to english locale for non-translated keys?

**Comment by huacnlee** [comment]
Yes, the translation is finished, you can merge it.

**Comment by huacnlee** [comment]
@simi I18n fallback enabled now.

**Comment by huacnlee** [comment]
BTW, the result of zh-CN page:

**Comment by dwradcliffe** [comment]
:+1:

**Comment by tw0517tw** [comment]
Is there any switch to select language manually?

**Comment by simi** [comment]
Looks like it is not. But you can do it manually now via locale attribute in url https://rubygems.org/?locale=nl. It will be good to add locales list into footer for example.

**Comment by huacnlee** [comment]
Its will use the browser language by default.

**Pull request merged**